### PR TITLE
Implement assign operators (#95)

### PR DESCRIPTION
**Pull request by oll3**
* Operators implemented: +=, -=, /=, *=, %=, &=, |=, ^=, <<=, >>=

* Add parser tests for new operators

### CODE REVIEW

File: src/lib/exec.rs [hunk 1]
```diff
@@ -41,6 +41,21 @@ pub struct InterpreterBuilder {
     global: Value,
 }
 
+fn exec_assign_op(op: &AssignOp, v_a: ValueData, v_b: ValueData) -> Gc<ValueData> {
```
Comment (jasonwilliams) [inline src/lib/exec.rs]: Should this be `Value` instead of `Gc<ValueData>` ?

Comment (oll3) [reply src/lib/exec.rs]: Missed that, but fixed now :)

File: src/lib/exec.rs [hunk 2]
```diff
@@ -265,6 +280,28 @@ impl Executor for Interpreter {
...
+                match a.def {
+                    ExprDef::Local(ref name) => {
+                        let v_r_a = self.environment.get_binding_value(&name);
+                        let v_a = (*v_r_a).clone();
+                        value = exec_assign_op(op, v_a, v_b);
+                        self.environment
+                            .set_mutable_binding(&name, value.clone(), true);
+                    }
+                    ExprDef::GetConstField(ref obj, ref field) => {
+                        let v_r_a = self.run(obj)?;
```
Comment (jorendorff) [inline src/lib/exec.rs]: The spec says the left hand side of compound assignment is evaluated first. In `f().x += g()`, first `f` is called, then `g`.

Comment (oll3) [reply src/lib/exec.rs]: Thanks for pointing out, will have a look at it later.

**Pull request merged**